src/index.js

```diff
@@ -79,33 +79,33 @@ export default function promiseMiddleware(config = {}) {
       });
 
       /*
-       * @function handleReject
-       * @description Dispatch the rejected action and return
-       * an error object. The error object is the original error
-       * that was thrown. The user of the library is responsible for
-       * best practices in ensure that they are throwing an Error object.
-       * @params reason The reason the promise was rejected
+       * @function transformFulfill
+       * @description Transforms a fulfilled value into a success object.
        * @returns {object}
        */
+      const transformFulfill = (value = null) => {
+        const resolvedAction = getAction(value, false);
+        return { value, action: resolvedAction };
+      };
+
+      /*
+       * @function handleReject
+       * @description Dispatch the rejected action.
+       * @returns {void}
+       */
       const handleReject = reason => {
         const rejectedAction = getAction(reason, true);
         dispatch(rejectedAction);
-        throw reason;
```

Comment:
I've separated out the functional transforms, which are now `transformFulfill` and the identity function, from the side effects, which are now `handleFulfill` and `handleReject`.

There's no need to throw here because nobody reads the value of this promise.